src/main/java/duke/component/Storage.java

```diff
@@ -0,0 +1,76 @@
+package duke.component;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class Storage {
+    private String filePath;
+
+    /**
+     * Creates a Storage with specified filePath.
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads lines from the specified filePath.
+     * @return
+     * @throws IOException
+     */
+    public String[] load() throws IOException {
+        ArrayList<String> lines = new ArrayList<>();
+        BufferedReader br = null;
+        try {
+            FileReader fr = new FileReader(this.filePath);
+            br = new BufferedReader(fr);
+            String line;
+            while ((line = br.readLine()) != null) {
+                lines.add(line);
+            }
+        } finally {
+            if (br != null) {
+                br.close();
+            }
+        }
+        return lines.toArray(new String[lines.size()]);
+    }
+
+    /**
+     * Writes lines into the specified filePath.
+     * @param tl
+     * @throws IOException
+     */
+    public void save(TaskList tl) throws IOException {
+        BufferedWriter bw = null;
+        try {
+            FileWriter fw = new FileWriter(this.filePath);
+            bw = new BufferedWriter(fw);
+            for (Task t : tl.getTasks()) {
```

Comment:
Personally, I prefer moving everything inside the for loop into a single method so as to reduce indentation levels. I believe it's worth some consideration, especially if you wish to convert this for loop as well as others into Streams.